lib/grid.rb

```diff
@@ -0,0 +1,15 @@
+class Grid
+  attr_reader :cells
+
+  def initialize(cells = {})
+    @cells = cells
+  end
+
+  def mark(cell:, value:)
+    Grid.new(cells.merge(cell => value))
```

Comment:
Trying out the idea of keeping the underlying data immutable. So far, it's made the player level a bit more complicated.

Reply:
Oh, that makes me understand the turn method that was worrying me. 

I think you should hold off on immutable data. With immutable state, some high level object has to hold on to the source of truth, and usually wraps it in a number of methods. You are using the `Game` object for that high level object. Publish/subscribe is a pattern that works for immutable data, or (dare I say) a singleton that manages state. The Game object, particularly with just a instance var, doesn't seem like a secure place to hang on to the current source of truth. 

It is kind of like you have defined a good mechanism for operating on the data with `Grid`, but the guts of it are hanging out anywhere else.

Reply:
Also, the `cells.merge(cell => value)` is kind of a leaky abstraction. As a new dev on this "project" I wouldn't feel confident using this thing.